src/main/java/core/basesyntax/service/impl/CsvFileWriterServiceImpl.java

```diff
@@ -0,0 +1,38 @@
+package core.basesyntax.service.impl;
+
+import core.basesyntax.db.Storage;
+import core.basesyntax.service.CsvFileWriterService;
+import java.io.BufferedWriter;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.util.Map;
+
+public class CsvFileWriterServiceImpl implements CsvFileWriterService {
+    private static final String FRUIT = "fruit";
+    private static final String QUANTITY = "quantity";
+    private static final String COMA_SEPARATOR = ",";
+
+    @Override
+    public void writeToFile(String path) {
+        try (BufferedWriter bufferedWriter = new BufferedWriter(new FileWriter(path))) {
+            writeFirstLine(bufferedWriter);
+            for (Map.Entry<String, Integer> data : Storage.FRUITS.entrySet()) {
+                String stringBuilder = data.getKey()
+                        + COMA_SEPARATOR
+                        + data.getValue()
+                        + System.lineSeparator();
+                bufferedWriter.write(stringBuilder);
+            }
```

Comment:
Let's create a separate separate service that is responsible for report creation and then pass the result of report generation to this writeToFile method and write in a single write() call :)